app/code/Magento/Sales/Model/Order/ShipmentDocumentFactory/ExtensionAttributesProcessor.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Sales\Model\Order\ShipmentDocumentFactory;
+
+use Magento\Framework\Api\SimpleDataObjectConverter;
+use Magento\Framework\Reflection\ExtensionAttributesProcessor as AttributesProcessor;
+use Magento\Sales\Api\Data\ShipmentCreationArgumentsExtensionInterface;
+use Magento\Sales\Api\Data\ShipmentCreationArgumentsInterface;
+use Magento\Sales\Api\Data\ShipmentExtensionFactory;
+use Magento\Sales\Api\Data\ShipmentInterface;
+
+/**
+ * Build and set extension attributes for shipment.
+ */
+class ExtensionAttributesProcessor
```

Comment:
Looks like we can reuse `\Magento\Framework\Api\ExtensionAttributesFactory` instead of this class